torch/cuda/memory.py

```diff
@@ -86,7 +89,7 @@ def empty_cache():
         torch._C._cuda_emptyCache()
 
 
-def memory_stats(device=None):
+def memory_stats(device: Optional[_device_t] = None) -> Dict[str, Any]:
```

Comment:
Do you really need to use `_device_t` here? In this file `device` shouldn't be shadowing the type name

Reply:
But type of `device`, is still either `torch.device` or `int` or `None`? Instead of repeating the same pattern, I've introduced `_optional_device_t` as union between 3 above-mentioned types

Reply:
Right, @rgommers and I have been discussing, we need a meta-type for all device like objects. But sort of similar to how Python naming convention works, I'd have always thought we ought to name it like Device (similar to how you have dict and Dict)

Reply:
Good point! Let me replace _device_t with https://github.com/pytorch/pytorch/blob/7c9e78fdf5765e6a7a08ac44dbf1ecd670afef09/torch/types.py#L31